crates/wit-parser/src/decoding.rs

```diff
@@ -20,6 +26,22 @@ struct ComponentInfo {
     types: types::Types,
     /// List of all imports and exports from this component.
     externs: Vec<(String, Extern)>,
+    /// Packages
+    explicit: Vec<ExplicitPackageInfo>,
+    /// Decoded package metadata
+    package_metadata: Option<PackageMetadata>,
+}
+
+/// Represents information about a decoded WebAssembly component.
+#[derive(Default)]
+struct ExplicitPackageInfo {
+    /// Package name
+    name: Option<PackageName>,
+    /// Wasmparser-defined type information learned after a component is fully
+    /// validated.
+    types: Option<types::Types>,
```

Comment:
It should be ok to only store the types at the "root" since I don't think a snapshot is needed along the way

Reply:
I see, at this point though I think I've been able to combine `ComponentInfo` with `ExplicitPackageInfo`, so I think it'll just have a value of `None` when for nested packages.